### PR TITLE
[TASK] Add basic POST support to executeFrontendRequest.

### PR DESCRIPTION
**Pull request by sven-carstens**
* The http method verb is now taken from the request instead of being fixed to GET
* body content is passed along to the request

### CODE REVIEW

**Comment by sven-carstens** [comment]
* request headers are now passed along to the inside

**Comment by maddy2101** [comment]
Can you give me some use case description? Why do you want this change? Probably a core patch that relies on this change, so please let me know in order to properly test the change.

**Comment by sven-carstens** [comment]
I want to do some functional tests for a middleware.
The middleware not only supports GET but also POST requests.
Without this patch the HTTP Verb is fixed to "GET" abd there is no way to get anything else to the tested middleware.

With the patch you can set other methods and you can also set the request body.

File: Classes/Core/Functional/Framework/Frontend/InternalRequest.php [hunk 1]
```diff
@@ -61,13 +61,20 @@ function (array $data) {
 
     /**
      * @param string|null $uri URI for the request, if any.
+     * @param string $method method to use, GET is default.
+     * @param string|null $content body content to use, if any.
+     * @param array $headers Headers for the message, if any.
      */
-    public function __construct($uri = null) {
+    public function __construct($uri = null, $method = 'GET', string $content = null, array $headers = []) {
```
Comment (bmack) [inline Classes/Core/Functional/Framework/Frontend/InternalRequest.php]: how about we'll add an (optional) stream instead of a string?

Comment (sven-carstens) [reply Classes/Core/Functional/Framework/Frontend/InternalRequest.php]: Whatever is not converted too many times firther down the road.

Comment (sven-carstens) [reply Classes/Core/Functional/Framework/Frontend/InternalRequest.php]: Strangely moving the
```
        $body = new Stream('php://temp', 'wb+');
        if (null !== $content) {
            $body->write($content);
            $body->rewind();
        }
```

to the outer class and changing the type of the parameter broke everything. The body does not get passed along to the ServerRequestFactory

**Comment by simonschaufi** [comment]
@sven-carstens @bmack can this PR be finished? I would love to use this feature!

File: Classes/Core/Functional/Framework/Frontend/InternalRequest.php (hunk 1 above)
Comment (sven-carstens) [reply Classes/Core/Functional/Framework/Frontend/InternalRequest.php:68]: > Strangely moving the
> 
> ```
>         $body = new Stream('php://temp', 'wb+');
>         if (null !== $content) {
>             $body->write($content);
>             $body->rewind();
>         }
> ```
> 
> to the outer class and changing the type of the parameter broke everything. The body does not get passed along to the ServerRequestFactory

Sorry, I give up on this.

**Pull request closed without merging**